app/domain/repos/conjur_ca.rb

```diff
@@ -14,12 +14,28 @@ def self.create(resource_id)
     #
     def self.ca(resource_id)
       ca_info = ::Conjur::CaInfo.new(resource_id)
-      stored_cert = Resource[ca_info.cert_id].last_secret.value
-      stored_key = Resource[ca_info.key_id].last_secret.value
+      stored_cert, stored_key = fetch_ca_secrets(ca_info, resource_id)
```

Comment:
i think this is quite risky, as it assumes the order of the retrieval. what happens if someone will change the behaviour of `FetchRequiredSecrets` so that it retrieves the secrets in parallel? In this case the first secret in the array may not be `stored_cert`. How can you overcome this?

Reply:
Changed